crates/generators/python/common.py

```diff
@@ -141,16 +142,34 @@ def generate_enum_property(
     return result
 
 
-def generate_enum(name: str, variants: list[str]) -> str:
+def generate_enum(name: str, variants: List[str]) -> str:
     """Generates an enum definition with the provided variants."""
-    body = ','.join(variants) + ','
+    return generate_enum_with_derives(name, variants, [])
+
+
+def generate_enum_with_derives(name: str, variants: List[str], derives: List[str]) -> str:
+    return generate_enum_with_derives_and_prelude(name, variants, derives, "")
+
 
-    return f"""
-    #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+def generate_enum_with_prelude(name: str, variants: List[str], prelude: str) -> str:
```

Comment:
We shouldn't have multiple different `generate_enum_with_x` methods. We could simply have a single `generate_enum` with optional parameters.

`def generate_enum(name: str, variants: List[str], derives: List[str] = [], prelude: str = "") -> str:`

Reply:
I originally created multiple functions because I didn't want to have to modify the block generator but I didn't think about functions with optional params. Good idea, thanks! 😄 Will do this evening.

Reply:
Done. Thanks! 😄